data/wikipron/src/scrape.py

```diff
@@ -58,10 +59,29 @@ def _build_scraping_config(
     phonetic_path = f"{path_affix}phonetic.tsv"
     _call_scrape(config_settings, phonetic_config, phonetic_path)
 
-
-def main() -> None:
+def main(args: argparse.Namespace) -> None:
     with open(LANGUAGES_PATH, "r") as source:
         languages = json.load(source)
+
+    codes = list(languages.keys())
+
+    # Verifies language code for --restriction is valid
+    if args.restriction:
+        # Cleans entry. 
+        keys = re.split(r'[;,\s]+\s*',args.restriction[0].strip(';, '))
+        if not keys[0]:
```

Comment:
I think you can say `nargs="+"` to require at least one and make this logic unnecessary?

Reply:
Actually, sorry to keep changing but I think also you can do something like:

    --restriction=fre --restriction=spa

to have multiple values. Then you don't need to do the `re`-splitting (and will no longer need `re`).

Reply:
The argparse is registering empty string cases even with nargs="+." So if --restriction= is passed then it will search for '' as a string and then log '' is not a valid ISO code. Which doesn't seem informative for an error.

Reply:
Ah, didn't realize that. Not very useful, I agree, keep this then.